src/main/java/com/linecorp/armeria/common/ServiceInvocationContext.java

```diff
@@ -244,6 +247,50 @@ public final SocketAddress remoteAddress() {
         return (T) originalRequest;
     }
 
+    /**
+     * Resolves the specified {@code promise} with the specified {@code result}. If {@code promise} is done
```

Comment:
Should explain what may cause a promise to be done already, it's very unexpected at first sight.

done already (e.g., it was fulfilled and returned by TimeoutHandler),